unittest/gunit/keyring/keyring-api-t.cc

```diff
@@ -223,6 +223,229 @@ namespace keyring__api_unittest
                               sample_key_data.length() + 1), 1);
   }
 
+  TEST_F(Keyring_api_test, StorePBStoreSKFetchPBRemovePB)
+  {
+    EXPECT_EQ(mysql_key_store("percona_binlog", "AES", NULL, sample_key_data.c_str(),
+                              sample_key_data.length() + 1), 0);
+    EXPECT_EQ(mysql_key_store("percona_RGRGRG_1", "AES", NULL, "1234_",
+                              strlen("1234_") + 1), 0);
+    char *key_type= NULL;
+    size_t key_len= 0;
+    void *key= NULL;
+    EXPECT_EQ(mysql_key_fetch("percona_binlog", &key_type, NULL, &key,
+                              &key_len), 0);
+    EXPECT_STREQ("AES", key_type);
+    std::string key_data_with_version= "0:" + sample_key_data;
+    EXPECT_EQ(key_len, key_data_with_version.length()+1);
+    ASSERT_TRUE(memcmp((char *)key, key_data_with_version.c_str(), key_len) == 0);
```

Comment:
C++-style casts here and below, but IMHO only update if you have to touch it otherwise

Reply:
Done